### DISCUSSION TITLE
Support for Luminea ZX-5140-*LED Controller

### DISCUSSION THREAD

**Discussion by ThomasCr** [arendst/Tasmota#13978] [Q&A] [answered]
Hi,

I search for information for Luminea ZX-5140-* WIFI LED Controller I ordered from [Amazon](https://www.amazon.de/gp/product/B09MKXSNDN).

It is easy to open, so I have made some detailed Pictures. It would be nice if it can be flashed with Tasmota.

The Controller have a Wi-Fi and Bluetooth LE combo chip [BK7231N](http://docs.bekencorp.com/v3.0/get-started/demo_board.html#bk7231n) from Beken Corporation.
 
[A Pin Description I have found on this website.](https://developer.tuya.com/en/docs/iot/tuya-sandwich-wifi-and-ble-soc-board-BK7231N?id=Kao72e6net3bs)

Tuya SDK for this Chip: https://github.com/tuya/tuya-iotos-embeded-sdk-wifi-ble-bk7231n

The controller has a build-in microphone to let the LEDs be controlled by the music. Also there is only a two wire system, but the LEDs can be controlled in alternating order (first is on, second is off and so on...)

Can this behavior also be set with Tasmota?

Can anybody guide me a little...

How to find the serial i/o pins?
What Firmware should I use?
How to identify the GPIO pins? ([I found this page, which I can follow, when I have flashed the device](https://tasmota.github.io/docs/Configuration-Procedure-for-New-Devices/#finding-relays-and-lights))

Case Front:
![IMG_20211209_193908~2](https://user-images.githubusercontent.com/226670/145459733-4a5593e2-1a83-4be8-b72a-999262534454.jpg)
Case Back:
![IMG_20211209_193840~2](https://user-images.githubusercontent.com/226670/145459776-526c85e4-172b-4578-a8a4-89f5076458a8.jpg)

board front left:
![IMG_20211209_193705](https://user-images.githubusercontent.com/226670/145460252-6decc75e-d11f-46e5-a9c3-59fb6a1d43af.jpg)
board front right:
![IMG_20211209_193638](https://user-images.githubusercontent.com/226670/145460344-61b2fe22-8439-4a05-9d70-32661d14a2ac.jpg)

board backside left:
![IMG_20211209_193729](https://user-images.githubusercontent.com/226670/145459869-893f4862-63a3-4e92-9499-b78299e7fb9a.jpg)
board backside right:
![IMG_20211209_193716](https://user-images.githubusercontent.com/226670/145460069-fab24a87-281a-4763-a3cd-bb001e975339.jpg)

**Comment by ascillato** [comment, marked as answer]
Sorry, Tasmota runs only in ESP8266, ESP8285 and ESP32 chips.

**Comment by ThomasCr** [comment]
thanks, I found a other project which is based on the Beken or Tuya SDK for this chip. But dont think that this is also be ready for mic usage and so on...

https://www.reddit.com/r/tasmota/comments/r58l75/wb2sbk7231t_cracked_sdk_setup_uart_firmware/

**Reply by Jason2866** [reply]
It is far away from being useable. It is a **lot** of work ahead. Too many for one.
Probably the chip will not be in devices when it is done...

**Reply by ThomasCr** [reply]
yes also think so....

**Reply by Dlay2017** [reply]
Now you can use OpenBeken to flash this device, to have a "tasmota" device

[https://www.elektroda.com/rtvforum/topic3951016.html](url)
